clang/tools/driver/cc1_main.cpp

```diff
@@ -235,6 +235,25 @@ namespace {
 //     handled.
 class CompileJobCache {
 public:
+  /// Categorization for the output kinds that is used to decouple the
+  /// compilation cache key from the specific output paths.
+  enum class OutputKind {
+    MainOutput,
+    SerializedDiagnostics,
```

Comment:
What about dependency file? When we go through the "full dependencies" API the dependency file is written by the compiler not the scanner.

Reply:
> When we go through the "full dependencies" API the dependency file is written by the compiler not the scanner.

Are there tests that exercise this? I thought all the caching tests we have are using the driver's caching mechanism.

Reply:
clang/test/CAS/fcache-compile-job-dependency-file.c uses cc1 commands, so it should exercise this without going through the driver/scanner.

Reply:
Handling dependency files as well in updated PR, using `test/CAS/fcache-compile-job-dependency-file.c` for testing.